table/block_based/block_based_table_iterator.h

```diff
@@ -250,11 +250,23 @@ class BlockBasedTableIterator : public InternalIteratorBase<Slice> {
   // BlockHandleInfo is used to store the info needed when block cache lookup
   // ahead is enabled to tune readahead_size.
   struct BlockHandleInfo {
-    BlockHandleInfo() {}
+    void SetFirstInternalKey(const Slice& key) {
+      size_t size = key.size();
```

Comment:
Index entry having the first key is optional, IIUC. If the key is empty, we can skip this.